client/my-sites/sharing/connections/connection.jsx

```diff
@@ -124,12 +126,13 @@ class SharingConnection extends Component {
 	}
 
 	getConnectionKeyringUserLabel() {
-		const keyringUser = site().getUser( this.props.connection.keyring_connection_user_ID );
+		const { connection, siteId, translate, userId } = this.props;
+		const keyringUser = site( { ID: siteId } ).getUser( connection.keyring_connection_user_ID );
```

Comment:
Indeed. If it's falsey there will be no `keyringUser` and the method doesn't return anything, like in cases when the connection was set up by the current user.